src/main.rs

```diff
@@ -65,6 +65,42 @@ struct Options<'a> {
     true_color: bool,
     style: OptionsStyle,
     language: Option<&'a str>,
+    interactive_terminal: bool,
+}
+
+enum OutputType<'a> {
+    Pager(Child),
+    Stdout(StdoutLock<'a>),
+}
+
+impl<'a> OutputType<'a> {
+    fn new_pager() -> Result<Self> {
+        Ok(OutputType::Pager(Command::new("less")
+            .args(&["--quit-if-one-screen", "--RAW-CONTROL-CHARS", "--no-init"])
+            .stdin(Stdio::piped())
+            .spawn()
+            .chain_err(|| "Could not spawn pager")?
+        ))
+    }
+
+    fn new_stdout(stdout: &'a Stdout) -> Self {
```

Comment:
I would prefer not to pass in the stdout, but I can't find a way around the lifetime issues